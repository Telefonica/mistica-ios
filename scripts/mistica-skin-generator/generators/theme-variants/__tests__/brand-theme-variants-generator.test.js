import fs from 'fs';
import {resolve} from 'path';
import {generateBrandThemeVariants} from '../brand-theme-variants-generator.js';

test('generateBrandThemeVariants', () => {
    const anyBrandTokensPath = resolve(__dirname, 'any-brand-tokens.json');
    const anyBrandTokens = JSON.parse(fs.readFileSync(anyBrandTokensPath, 'utf-8'));
    const brand = {prefix: 'O2New'};
    const result = generateBrandThemeVariants(brand, anyBrandTokens);

    expect(result).toMatchSnapshot();
});
