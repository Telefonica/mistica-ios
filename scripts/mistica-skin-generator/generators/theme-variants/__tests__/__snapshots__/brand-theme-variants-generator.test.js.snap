// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateBrandThemeVariants 1`] = `
"
// Generated using Make
// DO NOT EDIT

import Foundation

struct O2NewThemeVariants: MisticaThemeVariants {
    var successFeedback: MisticaThemeVariantType = .inverse
    var brandLoadingScreen: MisticaThemeVariantType = .inverse
}
"
`;
